index-scheduler/src/lib.rs

```
@@ -381,11 +424,21 @@ impl IndexScheduler {
         match wtxn.commit() {
             Ok(()) => (),
             _e @ Err(_) => {
-                todo!("remove the data associated with the task");
+                self.delete_persisted_task_data(&task)?;
                 // _e?;
```

We should throw the error after that